observatory-platform/observatory/platform/workflows/stream_telescope.py

```diff
@@ -261,43 +254,34 @@ def bq_delete_old(self, release: StreamRelease, **kwargs):
         """
         ti: TaskInstance = kwargs["ti"]
         if release.first_release:
-            # don't use AirflowSkipException, to ensure that task is in 'success' state, this is used below in
-            # ti.previous_start_date_success
-            logging.info("Skipped, because first release")
-            return
+            ti.xcom_push(key=self.XCOM_BQ_DATES, value=ti.start_date.strftime("%Y%m%d"))
```

Comment:
It would be good to not use XComs, especially for passing data between different DAG runs because it means that you can't just delete all of the XComs if they have problems.

In Airflow 2.2 there is a data interval property for each DAG run that tells you when the data period starts and ends. That isn't quite read yet, so in the PR https://github.com/The-Academic-Observatory/observatory-platform/pull/504 I've made a function called [get_data_interval](https://github.com/The-Academic-Observatory/observatory-platform/blob/e67c7f2431562ad4d2b6e63edd6deeb259904d46/observatory-platform/observatory/platform/workflows/stream_telescope.py#L70-L78) which does the same thing. The XComs have also been removed from [get_release_info](https://github.com/The-Academic-Observatory/observatory-platform/blob/e67c7f2431562ad4d2b6e63edd6deeb259904d46/observatory-platform/observatory/platform/workflows/stream_telescope.py#L158-L182). Once Airflow 2.2 comes out we can simplify the code a little bit.

Would you be able to do something similar in bq_delete_old and bq_append_new, using dag_run.get_previous_dagrun and get_data_interval?

Reply:
Thanks @jdddog!

I've removed the usage of xcoms.
The BigQuery related tasks need to know the date of the last time that task was in the 'success' state, so they can't use the get_previous_dagrun / get_data_interval directly.
I have added a function to find the start date of when the task was last successful.